Web Inspector: Fonts: Warning text for synthesized oblique/italic should mention the correct property value

rcaliman-apple opened the pull request:
#### 16546e0d598306118d9440a1ffd5f35aa69d36ef
<pre>
Web Inspector: Fonts: Warning text for synthesized oblique/italic should mention the correct property value
<a href="https://bugs.webkit.org/show_bug.cgi?id=254759">https://bugs.webkit.org/show_bug.cgi?id=254759</a>

Reviewed by Devin Rousso.

Show a warning message corresponding to the style that didn&apos;t have font support and was synthesized: italic vs oblique.

* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
* Source/WebInspectorUI/UserInterface/Views/FontDetailsPanel.js:
(WI.FontDetailsPanel.prototype.update):

Canonical link: <a href="https://commits.webkit.org/262426@main">https://commits.webkit.org/262426@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c199df3bb95978f93e39814d3d3eb4b4f274666f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1460 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1490 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1540 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2386 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/1291 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1551 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1550 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1409 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1473 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/1338 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1321 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2225 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1347 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1304 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1264 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1332 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1344 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2420 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1385 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/1231 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1303 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1311 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/374 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1427 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/FontDetailsPanel.js:
```diff
@@ -76,7 +76,10 @@ WI.FontDetailsPanel = class FontDetailsPanel extends WI.StyleDetailsPanel
                 if (propertyName === "font-style")
-                    row.warningMessage = this.nodeStyles.computedPrimaryFont?.synthesizedOblique ? WI.UIString("Font was synthesized to be oblique because no oblique font is available.", "A warning that is shown in the Font Details Sidebar when the font had to be synthesized to support the provided style.") : null;
+                    if (this.nodeStyles.computedPrimaryFont?.synthesizedOblique)
+                        row.warningMessage = fontProperty.value === "italic" ? WI.UIString("Font was synthesized to be italic because no italic font is available.", "A warning that is shown in the Font Details Sidebar when the font had to be synthesized to support the provided style.") : WI.UIString("Font was synthesized to be oblique because no oblique font is available.", "A warning that is sho ...
```
NIT: we may want this to be a 
```
if (fontProperty.value === "italic")
    row.warningMessage = ...
else
    row.warningMessage = ...
```
just so that it's not such a long line :)

patrickangle replied:
Isn't the italic string still wrong? In both cases oblique-ness is synthesized, regardless of what was requested IIRC. I'd think `"Font was synthesized to be oblique because no italic font is available."` would be more accurate?

rcaliman-apple replied:
> Isn't the italic string still wrong? In both cases oblique-ness is synthesized, regardless of what was requested IIRC. I'd think `"Font was synthesized to be oblique because no italic font is available."` would be more accurate?

You're right! We discussed this exact thing and I forgot. 
Updated the warning message for `italic`.

<img width="572" alt="Screenshot 2023-03-31 at 16 09 32" src="https://user-images.githubusercontent.com/105809798/229145563-8c9811fc-7336-40d4-90b9-67d1f0af63f0.png">

rcaliman-apple committed:
Web Inspector: Fonts: Warning text for synthesized oblique/italic should mention the correct property value
https://bugs.webkit.org/show_bug.cgi?id=254759

Reviewed by Devin Rousso.

Show a warning message corresponding to the style that didn't have font support and was synthesized: italic vs oblique.

* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
* Source/WebInspectorUI/UserInterface/Views/FontDetailsPanel.js:
(WI.FontDetailsPanel.prototype.update):

Canonical link: https://commits.webkit.org/262426@main
